tests/suites/test_suite_pk.function

```diff
@@ -1924,3 +1968,202 @@ exit:
+
+exit:
+    mbedtls_pk_free(&pk_ctx);
+    psa_destroy_key(key_id);
+    PSA_DONE();
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_PSA_CRYPTO_CLIENT*/
+void pk_copy_from_psa_success(data_t *priv_key_data, int key_type_arg,
+                              int key_bits_arg, int key_usage_arg,
+                              int key_alg_arg)
+{
+    psa_key_type_t key_type = key_type_arg;
+    psa_key_type_t pub_key_type;
+    size_t key_bits = key_bits_arg;
+    psa_key_usage_t key_usage = key_usage_arg;
+    psa_algorithm_t key_alg = key_alg_arg;
+    mbedtls_pk_context pk_ctx, pk_ctx2;
+    mbedtls_svc_key_id_t priv_key_id = MBEDTLS_SVC_KEY_ID_INIT;
+    mbedtls_svc_key_id_t pub_key_id = MBEDTLS_SVC_KEY_ID_INIT;
+    unsigned char *in_buf = NULL;
+    size_t in_buf_len = MBEDTLS_MD_MAX_SIZE;
+    unsigned char out_buf[MBEDTLS_PK_SIGNATURE_MAX_SIZE];
+    unsigned char out_buf2[MBEDTLS_PK_SIGNATURE_MAX_SIZE];
+    size_t out_buf_len, out_buf2_len;
+
+    /* Get the MD type to be used for the tests below from the provided key policy. */
+    mbedtls_md_type_t md_for_test = MBEDTLS_MD_SHA256; /* Default */
+    if ((PSA_ALG_GET_HASH(key_alg) != 0) &&
+        (PSA_ALG_GET_HASH(key_alg) != PSA_ALG_ANY_HASH)) {
+        md_for_test = mbedtls_md_type_from_psa_alg(key_alg);
+    }
+
+    in_buf_len = mbedtls_md_get_size_from_type(md_for_test);
+    TEST_CALLOC(in_buf, in_buf_len);
+    memset(in_buf, 0x1, in_buf_len);
+
+    mbedtls_pk_init(&pk_ctx);
+    mbedtls_pk_init(&pk_ctx2);
+    PSA_INIT();
+
+    /* Generate a private key in PSA and create a PK context from it. */
+    PSA_ASSERT(pk_psa_import_key(priv_key_data->x, priv_key_data->len,
+                                 key_type, key_usage, key_alg, key_bits, &priv_key_id));
+    TEST_EQUAL(mbedtls_pk_copy_from_psa(priv_key_id, &pk_ctx), 0);
+
+    /* Starting from the private key above, create another PSA slot for the public
+     * one and create a new PK context from it. */
+    if (PSA_KEY_TYPE_IS_ECC_KEY_PAIR(key_type)) {
+        pub_key_type = PSA_KEY_TYPE_ECC_PUBLIC_KEY(PSA_KEY_TYPE_ECC_GET_FAMILY(key_type));
+    } else if (key_type == PSA_KEY_TYPE_RSA_KEY_PAIR) {
+        pub_key_type = PSA_KEY_TYPE_RSA_PUBLIC_KEY;
+    } else {
+        TEST_FAIL("Key type can only be EC or RSA key pair");
+    }
+
+    /* Generate a 2nd PK contex using only the public key derived from its private
+     * counterpart generated above.  */
+    pub_key_id = pk_psa_pub_key_from_priv(priv_key_id, pub_key_type, key_usage, key_alg, key_bits);
+    TEST_EQUAL(mbedtls_pk_copy_from_psa(pub_key_id, &pk_ctx2), 0);
+
+    /* Test sign/verify with the following parttern:
+     * - Sign using the PK context generated from the private key.
+     * - Verify from the same PK context used for signature.
+     * - Verify with the PK context generated using public key.
+     */
+    if ((PSA_ALG_IS_RSA_OAEP(key_alg) || PSA_ALG_IS_RSA_PSS(key_alg))) {
```

Comment:
Note: ideally we want tests with the non-ext functions in this case as well. I think we still need a bit more discussion about what should be possible with what original algs (and what it should do).